sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -9,54 +10,126 @@ import com.klaviyo.coresdk.networking.requests.KlaviyoRequest
 import com.klaviyo.coresdk.networking.requests.TrackRequest
 
 /**
- * Our main API for the Klaviyo SDK.
- * Where we pass in analytics requests to be sent to the Klaviyo backend
+ * Public API for the core Klaviyo SDK.
+ * Receives configuration, customer data, and analytics requests
+ * to be processed and sent to the Klaviyo backend
  */
 object Klaviyo {
 
     /**
-     * Assigns a user email address to the user info
-     * This email address is stored in memory to define user details during a session and will be
-     * used to autocomplete analytics requests that are missing the identifier in their property maps
+     * Configure Klaviyo SDK with your account's public API Key and application context.
+     * Optionally specify additional behavior customization
      *
-     * Whenever the active user in an app changes (for example via a fresh login) this should be
-     * called to set the new user information in memory.
+     * This must be called to initialize the SDK before using any other functionality
      *
-     * @param email Email address for an active user
+     * @param apiKey - Your Klaviyo account's public API Key
+     * @param applicationContext
+     * @param networkTimeout
+     * @param networkFlushInterval
+     * @param networkFlushDepth
+     * @param networkFlushCheckInterval
+     * @param networkUseAnalyticsBatchQueue
+     * @return
      */
-    fun setUserEmail(email: String) {
+    fun configure(
+        apiKey: String,
+        applicationContext: Context,
+        networkTimeout: Int = KlaviyoConfig.NETWORK_TIMEOUT_DEFAULT,
```

Comment:
Interesting we are allow these be configured.

Reply:
IDK if this will survive, it was publicly exposed before, so I kept it public for now.

Reply:
I vaguely remember some ancient conversation around this years ago. There was some reason we had them exposed at the time but it's likely overkill and something we can just make private before we launch

Reply:
Yah I don't have it yet...also not sure we'd want folks mucking with it.

Reply:
Or making it easy to muck with it.

Reply:
BTW these parameters were the reason the tester app only sends any API calls after pressing the button twice...  `networkFlushDepth` was set to 2, when I changed it to 1 it was fine

Reply:
We might want to adjust that higher for production.

Reply:
For sure. But that was an override specified from the test-app code.

Reply:
If an app developer knows that they are building and distributing their app in regions where network latency is a concern would it make sense for them to configure a large time out?

Reply:
Could be an argument for keeping it configurable. On the other hand, I don't think I've ever encountered an SDK of this sort that allowed this level of customization.

Reply:
I'm marking this thread resolved but making a note of this convo in the android arch document